### PR TITLE
[AutoDiff] NFC: Rename SILReverseAutoDiff* to SILAutoDiff*.

### PR DESCRIPTION
**Pull request by rxwei**
As title. No need to differentiate (no pun intended) between forward and reverse.

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow linux

File: include/swift/SIL/SILNodes.def [hunk 1]
```diff
@@ -620,6 +620,10 @@ ABSTRACT_VALUE_AND_INST(SingleValueInstruction, ValueBase, SILInstruction)
   // Automatic differentiation
   SINGLE_VALUE_INST(GradientInst, gradient,
                     SingleValueInstruction, None, DoesNotRelease)
+  SINGLE_VALUE_INST(AutoDiffFunction, autodiff_function,
+                    SingleValueInstruction, None, DoesNotRelease)
+  SINGLE_VALUE_INST(AutoDiffFunctionExtract, autodiff_function_extract,
+                    SingleValueInstruction, None, DoesNotRelease)
```
Comment (marcrasi) [inline include/swift/SIL/SILNodes.def]: Are these additions intentional?

**Review by marcrasi — APPROVED** [review]

File: include/swift/SIL/SILNodes.def (hunk 1 above)
Comment (rxwei) [reply include/swift/SIL/SILNodes.def]: non-intentional. thanks for the catch.

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@shahmishal Mishal, I think something's wrong with update-checkout in CI. Could you take a look?

**Comment by marcrasi** [comment]
Maybe a clean test will fix it? I'll try triggering a clean test.

**Comment by marcrasi** [comment]
@swift-ci please clean test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Pull request merged**